### PR TITLE
ToricVarieties: Workaround for bug in polymake in weighted_projective…

### LINKED ISSUE #1687: ToricVarietiies: Shuffling weights for weighted projective space changes the space
```julia
julia> matrix(ZZ,rays(weighted_projective_space(NormalToricVariety, [2,3,1])))
[-3   -1]
[ 1    0]
[ 0    1]

julia> matrix(ZZ,rays(weighted_projective_space(NormalToricVariety, [1,2,3])))
[-2   -3]
[ 1    0]
[ 0    1]
```
Apparently, the two fans and the associated toric varieties are not isomorphic. However, they should be (if not identical by construction).

This method relies on `Polymake.fulton.weighted_projective_space(w)` where `w` are the weights in question. @lkastner what do you think?

Further background: `P_{231}`, i.e. `weighted_projective_space(NormalToricVariety, [2,3,1])` is the natural habitat for elliptic curves and elliptic fibrations, which I use daily for my physics applications. [As quick/hot fix, I will work with `P_{123}` for now.]

### PR DESCRIPTION
**Pull request by lkastner**
Closes #1687

### CODE REVIEW

File: src/ToricVarieties/NormalToricVarieties/constructors.jl [hunk 1]
```diff
@@ -316,8 +316,31 @@ function weighted_projective_space(::Type{NormalToricVariety}, w::Vector{T}) whe
...
+        lattice_gens[i+1,i] = div(lcms[i], w[i+1])
+        ray_gens[i+1,i] = lcms[i]
+    end
+    H = hnf(lattice_gens)
+    lattice_gens = transpose(H[1:length(w)-1, :])
+    for i in 1:nrows(ray_gens)
+        ray_gens[i,:] = transpose(solve(lattice_gens, transpose(ray_gens[i,:])))
+    end
+    mc = IncidenceMatrix(subsets(Vector{Int}(1:length(w)), length(w)-1))
+    variety = NormalToricVariety(PolyhedralFan(ray_gens, mc))
```
Comment (HereAround) [inline src/ToricVarieties/NormalToricVarieties/constructors.jl]: Thank you @lkastner for the quick fix. I am mostly happy but would appreciate if we could fix the order of the rays such that their order reflects the order of the entered weights. Maybe `non_redundant = true` does the trick, i.e.
```suggestion
    variety = NormalToricVariety(PolyhedralFan(ray_gens, mc; non_redundant = non_redundant))
```

**Review by HereAround — CHANGES_REQUESTED** [review]

File: src/ToricVarieties/NormalToricVarieties/constructors.jl (hunk 1 above)
Comment (HereAround) [reply src/ToricVarieties/NormalToricVarieties/constructors.jl]: I see you have this now included. So this is resolved.

**Review by HereAround — APPROVED** [review]
Thank you @lkastner.

File: src/ToricVarieties/NormalToricVarieties/constructors.jl (hunk 1 above)
Comment (lkastner) [reply src/ToricVarieties/NormalToricVarieties/constructors.jl]: I already said this in the slack, but whatever.

**Review by benlorenz — APPROVED** [review]

**Pull request merged**